iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/persistence/pipe/PipeTaskInfo.java

```diff
@@ -279,18 +289,22 @@ public boolean hasExceptions(String pipeName) {
   }
 
   /**
-   * Clear the exceptions of a pipe locally after it starts successfully. If there are exceptions
-   * cleared, the messages will then be updated to all the nodes through
-   * PipeHandleMetaChangeProcedure.
+   * Clear the exceptions of, and set the isAutoStopped flag to false for a pipe locally after it
+   * starts successfully. If there are exceptions cleared or flag changed, the messages will then be
+   * updated to all the nodes through PipeHandleMetaChangeProcedure.
    *
    * @param pipeName The name of the pipes to be clear exception
    */
-  public void clearExceptions(String pipeName) {
+  public void clearExceptionsAndSetAutoStoppedToFalse(String pipeName) {
     if (!pipeMetaKeeper.containsPipeMeta(pipeName)) {
       return;
     }
     PipeRuntimeMeta runtimeMeta = pipeMetaKeeper.getPipeMeta(pipeName).getRuntimeMeta();
     runtimeMeta.setExceptionsClearTime(System.currentTimeMillis());
+
+    // Clear the isAutoStopped flag as well, to avoid unnecessary retry.
+    runtimeMeta.setIsAutoStopped(false);
```

Comment:
It's best to put them before 

```
runtimeMeta.setExceptionsClearTime(System.currentTimeMillis());
```

This will make 

```
runtimeMeta.setExceptionsClearTime(System.currentTimeMillis());
```
and 
```
Map<Integer, PipeRuntimeException> exceptionMap =
        runtimeMeta.getDataNodeId2PipeRuntimeExceptionMap();
    if (!exceptionMap.isEmpty()) {
      exceptionMap.clear();
    }
    runtimeMeta
        .getConsensusGroupId2TaskMetaMap()
        .values()
        .forEach(
            pipeTaskMeta -> {
              if (pipeTaskMeta.getExceptionMessages().iterator().hasNext()) {
                pipeTaskMeta.clearExceptionMessages();
              }
            });
```

side by side (they are all related to exceptions, but setIsAutoStopped is not).